src/modules/shopPage/components/mobileShopHeaderBar/Filter.tsx

```diff
@@ -0,0 +1,396 @@
+    },
+    {
+      id: 'brand',
+      options: filtersParams?.brands,
+      label: t('brand'),
+      onChange: handleAutocompleteChange('brand'),
+    },
+    {
+      id: 'studed',
+      options: [t('studded'), t('studless')],
+      label: t('studded'),
+      onChange: handleAutocompleteChange('studed'),
+    },
+  ];
+
+  return (
+    <>
+      <Button
+        onClick={toggleDrawer(true)}
+        variant="text"
+        sx={{
+          padding: '0px',
+          margin: '0px',
+          '@media (max-width: 918px)': {
+            minWidth: 0,
+          },
+        }}>
+        <MenuIcon
+          fontSize="large"
+          sx={{
+            color: 'white',
+          }}
+        />
+        <Typography
+          variant="subtitle1"
+          sx={{ color: 'white', marginTop: '0.2rem', paddingLeft: '0.4rem' }}>
+          {t('filters')}
+        </Typography>
+      </Button>
+      <SwipeableDrawer
+        anchor={'left'}
+        open={openFilter.left}
+        onClose={toggleDrawer(false)}
+        onOpen={toggleDrawer(true)}>
+        <Box
+          width={'50vw'}
+          role="presentation"
+          sx={{ '@media (max-width: 600px)': { width: '100vw' } }}
+          m={'5%'}>
+          <List>
+            {autocompleteOptions.map(({ id, options, label, onChange }) => (
+              <ListItem>
+                <StyledAutocomplete
+                  key={id}
+                  disablePortal
+                  onChange={onChange}
+                  options={
+                    id === 'width' || id === 'brand'
+                      ? sortOptions(options)
+                      : options
+                  }
+                  value={getValueForAutocomplete(id)}
+                  renderInput={(params) => (
+                    <TextField
+                      label={t(label)}
+                      {...params}
+                      InputLabelProps={{
+                        ...params.InputLabelProps,
+                        children: undefined,
+                      }}
+                    />
+                  )}
+                  sx={{
+                    width: '90%',
+                  }}
+                />
+              </ListItem>
+            ))}
+            <Box display={'flex'} width={'91%'} justifyContent={'space-around'}>
+              <Button
+                variant="contained"
+                onClick={handleCleareAllFilters}
+                sx={{
+                  mt: '1rem',
+                  width: '40%',
+                  height: '50px',
+                  fontFamily: FONTS.BOLD_TEXT_FAMILY,
+                  fontWeight: 'bold',
+                  color: FILTER_COLORS.TEXT_MAIN,
+                  background: FILTER_COLORS.SHORT_MENU_RESET_BUTTON_BACKGROUND,
+                  fontSize: '1rem',
+                  '&:hover': {
+                    background:
+                      FILTER_COLORS.SHORT_MENU_RESET_BUTTON_BACKGROUND,
+                  },
+                }}>
```

Comment:
if u have a component with the same styles like (fontFamily, color, bgcolor, hover event) you can move it to the shared folder as 'CommonButtonName'